include/xtensor/xtensor.hpp

```diff
@@ -518,6 +518,32 @@ namespace xt
     {
         return m_storage;
     }
+
+    /**
+     * Converts `std::vector<index_type>` (returned e.g. from `xt::argwhere`) to `xtensor`.
+     * @param vector of indices
+     * @return `xt::tensor<typename index_type::value_type, 2>` (e.g. `xt::tensor<size_t, 2>`)
+     */
+    template <class T>
+    inline auto from_indices(const std::vector<T> &idx)
+    {
+        if ( idx.size() == 0 )
+        {
+            return empty<typename T::value_type>({std::size_t(0),std::size_t(0)});
+        }
+
+        xtensor<typename T::value_type, 2> out = empty<typename T::value_type>({idx.size(),idx[0].size()});
+
+        for ( std::size_t i = 0 ; i < out.shape()[0] ; ++i )
+        {
+            for ( std::size_t j = 0 ; j < out.shape()[1] ; ++j )
+            {
+                out(i,j) = idx[i][j];
```

Comment:
My code formatting OCD speaking here 😏

- always a white space after `,`. For example `out(i, j)`, `{std::size_t(0), std::size_t(0)}`
- no whitespace before `;`
- no whitespaces within control statement parentheses `if (idx.size() == 0)`

Sorry for the nitpicking :)

Reply:
Other nitpicking, maybe doing

```
using return_type = xtensor<typename T::value_type, 2>;
using size_type = return_type::size_type;
```

then loop over `size_type` instead of `std::size_t`.

Reply:
Specifying the `return_type` at the top, and doing `return_type out = empty(...` would be less ambiguous.

Reply:
Thanks! Changed all.